code/GraphQL/LinkablePlugin.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+
+namespace SilverStripe\CMS\GraphQL;
+
+use GraphQL\Type\Definition\ResolveInfo;
+use SilverStripe\CMS\Model\SiteTree;
+use SilverStripe\Core\Config\Configurable;
+use SilverStripe\Core\Injector\Injectable;
+use SilverStripe\GraphQL\Schema\DataObject\Plugin\QueryFilter\QueryFilter;
+use SilverStripe\GraphQL\Schema\Field\ModelQuery;
+use SilverStripe\GraphQL\Schema\Interfaces\ModelQueryPlugin;
+use SilverStripe\GraphQL\Schema\Schema;
+use SilverStripe\ORM\DataList;
+
+if (!interface_exists(ModelQueryPlugin::class)) {
+    return;
+}
+
+class LinkablePlugin implements ModelQueryPlugin
+{
+    use Configurable;
+    use Injectable;
+
+    const IDENTIFIER = 'getByLink';
+
+    /**
+     * @var string
+     * @config
+     */
+    private static $field_name = 'link';
+
+    /**
+     * @return string
+     */
+    public function getIdentifier(): string
+    {
+        return self::IDENTIFIER;
+    }
+
+    public function apply(ModelQuery $query, Schema $schema, array $config = []): void
+    {
+        $class = $query->getModel()->getSourceClass();
+        // Only site trees have the get_by_link capability
+        if ($class !== SiteTree::class && !is_subclass_of($class, SiteTree::class)) {
+            return;
+        }
+        // if the query is intended to return a list, `link` doesn't apply here
+        if ($query->isList()) {
+            return;
```

Comment:
Shouldn't those two conditionals throw instead? Silent errors are the devil.

Reply:
Yup, fixed.